src/contract.rs

```diff
@@ -914,9 +1014,163 @@ fn query_validators_metrics_by_timestamp(
     Ok(res)
 }
 
+// off chain code
+
+//todo: test, validate
+fn remove_off_chain_metrics_for_timestamp(
+    deps: DepsMut,
+    info: MessageInfo,
+    timestamp: u64,
+) -> Result<Response, ContractError> {
+    let mut timestamp_existed = true;
+
+    if !sender_is_manager(&deps, &info) {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    if !OFF_CHAIN_TIMESTAMPS.has(deps.storage, U64Key::from(timestamp)) {
+        timestamp_existed = false;
+    }
+
+    OFF_CHAIN_TIMESTAMPS.remove(deps.storage, U64Key::from(timestamp));
+    OFF_CHAIN_TIMESTAMP_META_DATA.remove(deps.storage, U64Key::from(timestamp));
+
+    let validator_idxs_to_remove: Vec<u16> = OFF_CHAIN_STATE_FOR_VALIDATOR
+        .prefix(U64Key::from(timestamp))
+        .range(deps.storage, Option::None, Option::None, Order::Ascending)
```

Comment:
Can we paginate or delete say 30 validators at a time?
I believe this line will make the whole tx timeout.